src/azure-cli/azure/cli/command_modules/keyvault/_completers.py

```diff
@@ -4,21 +4,14 @@
 # --------------------------------------------------------------------------------------------
 
 from azure.cli.core.decorators import Completer
-from azure.cli.core._profile import Profile
-
-
-def _get_token(cli_ctx, server, resource, scope):  # pylint: disable=unused-argument
-    return Profile(cli_ctx=cli_ctx).get_login_credentials(resource)[0]._token_retriever()  # pylint: disable=protected-access
+from azure.cli.command_modules.keyvault._client_factory import keyvault_data_plane_factory
```

Comment:
As pointed by #15854, calling the protected method `_token_retriever` is simply wrong. 

Switch to calling `azure.cli.command_modules.keyvault._client_factory.keyvault_data_plane_factory` instead which calls the correct `get_raw_token` method provided by https://github.com/Azure/azure-cli/issues/15805#issuecomment-722914070.